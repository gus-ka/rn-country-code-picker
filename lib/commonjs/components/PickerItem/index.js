"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _react = _interopRequireDefault(require("react"));
var _reactNative = require("react-native");
var _Flag = _interopRequireDefault(require("../Flag"));
var _StyledText = _interopRequireDefault(require("../StyledText"));
var _styles = _interopRequireDefault(require("./styles"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
const PickerItem = _ref => {
  let {
    country,
    onCountrySelect,
    textStyle,
    containerStyle
  } = _ref;
  const {
    name,
    flag,
    callingCode
  } = country;
  return /*#__PURE__*/_react.default.createElement(_reactNative.TouchableOpacity, {
    onPress: () => onCountrySelect(country),
    testID: "picker-item"
  }, /*#__PURE__*/_react.default.createElement(_reactNative.View, {
    style: containerStyle ?? _styles.default.container
  }, /*#__PURE__*/_react.default.createElement(_reactNative.View, {
    style: _styles.default.imageContainer
  }, /*#__PURE__*/_react.default.createElement(_Flag.default, {
    flag: flag
  })), /*#__PURE__*/_react.default.createElement(_reactNative.View, {
    style: _styles.default.codeContainer
  }, /*#__PURE__*/_react.default.createElement(_StyledText.default, {
    style: textStyle
  }, `+${callingCode}`)), /*#__PURE__*/_react.default.createElement(_StyledText.default, {
    style: textStyle ?? _styles.default.textStyle
  }, name)));
};
var _default = PickerItem;
exports.default = _default;
//# sourceMappingURL=index.js.map