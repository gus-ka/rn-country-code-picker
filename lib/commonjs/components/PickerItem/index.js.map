{"version":3,"names":["_react","_interopRequireDefault","require","_reactNative","_Flag","_StyledText","_styles","obj","__esModule","default","PickerItem","_ref","country","onCountrySelect","textStyle","containerStyle","name","flag","callingCode","createElement","TouchableOpacity","onPress","testID","View","style","styles","container","imageContainer","codeContainer","_default","exports"],"sources":["index.tsx"],"sourcesContent":["import React from 'react';\nimport { TouchableOpacity, View } from 'react-native';\nimport { IPickerItemProps } from '../../types';\nimport Flag from '../Flag';\nimport StyledText from '../StyledText';\nimport styles from './styles';\n\nconst PickerItem: React.FC<IPickerItemProps> = ({\n  country,\n  onCountrySelect,\n  textStyle,\n  containerStyle,\n}) => {\n  const { name, flag, callingCode } = country;\n  return (\n    <TouchableOpacity onPress={() => onCountrySelect(country)} testID=\"picker-item\">\n      <View style={containerStyle ?? styles.container}>\n        <View style={styles.imageContainer}>\n          <Flag flag={flag} />\n        </View>\n        <View style={styles.codeContainer}>\n          <StyledText style={textStyle}>{`+${callingCode}`}</StyledText>\n        </View>\n        <StyledText style={textStyle ?? styles.textStyle}>{name}</StyledText>\n      </View>\n    </TouchableOpacity>\n  );\n};\n\nexport default PickerItem;\n"],"mappings":";;;;;;AAAA,IAAAA,MAAA,GAAAC,sBAAA,CAAAC,OAAA;AACA,IAAAC,YAAA,GAAAD,OAAA;AAEA,IAAAE,KAAA,GAAAH,sBAAA,CAAAC,OAAA;AACA,IAAAG,WAAA,GAAAJ,sBAAA,CAAAC,OAAA;AACA,IAAAI,OAAA,GAAAL,sBAAA,CAAAC,OAAA;AAA8B,SAAAD,uBAAAM,GAAA,WAAAA,GAAA,IAAAA,GAAA,CAAAC,UAAA,GAAAD,GAAA,KAAAE,OAAA,EAAAF,GAAA;AAE9B,MAAMG,UAAsC,GAAGC,IAAA,IAKzC;EAAA,IAL0C;IAC9CC,OAAO;IACPC,eAAe;IACfC,SAAS;IACTC;EACF,CAAC,GAAAJ,IAAA;EACC,MAAM;IAAEK,IAAI;IAAEC,IAAI;IAAEC;EAAY,CAAC,GAAGN,OAAO;EAC3C,oBACEZ,MAAA,CAAAS,OAAA,CAAAU,aAAA,CAAChB,YAAA,CAAAiB,gBAAgB;IAACC,OAAO,EAAEA,CAAA,KAAMR,eAAe,CAACD,OAAO,CAAE;IAACU,MAAM,EAAC;EAAa,gBAC7EtB,MAAA,CAAAS,OAAA,CAAAU,aAAA,CAAChB,YAAA,CAAAoB,IAAI;IAACC,KAAK,EAAET,cAAc,IAAIU,eAAM,CAACC;EAAU,gBAC9C1B,MAAA,CAAAS,OAAA,CAAAU,aAAA,CAAChB,YAAA,CAAAoB,IAAI;IAACC,KAAK,EAAEC,eAAM,CAACE;EAAe,gBACjC3B,MAAA,CAAAS,OAAA,CAAAU,aAAA,CAACf,KAAA,CAAAK,OAAI;IAACQ,IAAI,EAAEA;EAAK,CAAE,CACf,CAAC,eACPjB,MAAA,CAAAS,OAAA,CAAAU,aAAA,CAAChB,YAAA,CAAAoB,IAAI;IAACC,KAAK,EAAEC,eAAM,CAACG;EAAc,gBAChC5B,MAAA,CAAAS,OAAA,CAAAU,aAAA,CAACd,WAAA,CAAAI,OAAU;IAACe,KAAK,EAAEV;EAAU,GAAG,IAAGI,WAAY,EAAc,CACzD,CAAC,eACPlB,MAAA,CAAAS,OAAA,CAAAU,aAAA,CAACd,WAAA,CAAAI,OAAU;IAACe,KAAK,EAAEV,SAAS,IAAIW,eAAM,CAACX;EAAU,GAAEE,IAAiB,CAChE,CACU,CAAC;AAEvB,CAAC;AAAC,IAAAa,QAAA,GAEanB,UAAU;AAAAoB,OAAA,CAAArB,OAAA,GAAAoB,QAAA"}