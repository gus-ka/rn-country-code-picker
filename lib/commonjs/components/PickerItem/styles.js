"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _reactNative = require("react-native");
var _theme = require("../../theme");
var _default = _reactNative.StyleSheet.create({
  container: {
    flexDirection: 'row',
    padding: _theme.spacing.s
  },
  imageContainer: {
    alignSelf: 'center'
  },
  codeContainer: {
    alignSelf: 'center'
  },
  textStyle: {
    alignSelf: 'center'
  }
});
exports.default = _default;
//# sourceMappingURL=styles.js.map