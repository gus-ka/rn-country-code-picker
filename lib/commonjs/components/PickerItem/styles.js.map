{"version":3,"names":["_reactNative","require","_theme","_default","StyleSheet","create","container","flexDirection","padding","spacing","s","imageContainer","alignSelf","codeContainer","textStyle","exports","default"],"sources":["styles.ts"],"sourcesContent":["import { StyleSheet } from 'react-native';\nimport { colors, spacing } from '../../theme';\n\nexport default StyleSheet.create({\n  container: {\n    flexDirection: 'row',\n    padding: spacing.s,\n  },\n  imageContainer: {\n    alignSelf: 'center',\n  },\n  codeContainer: {\n    alignSelf: 'center',\n  },\n  textStyle: { alignSelf: 'center' },\n});\n"],"mappings":";;;;;;AAAA,IAAAA,YAAA,GAAAC,OAAA;AACA,IAAAC,MAAA,GAAAD,OAAA;AAA8C,IAAAE,QAAA,GAE/BC,uBAAU,CAACC,MAAM,CAAC;EAC/BC,SAAS,EAAE;IACTC,aAAa,EAAE,KAAK;IACpBC,OAAO,EAAEC,cAAO,CAACC;EACnB,CAAC;EACDC,cAAc,EAAE;IACdC,SAAS,EAAE;EACb,CAAC;EACDC,aAAa,EAAE;IACbD,SAAS,EAAE;EACb,CAAC;EACDE,SAAS,EAAE;IAAEF,SAAS,EAAE;EAAS;AACnC,CAAC,CAAC;AAAAG,OAAA,CAAAC,OAAA,GAAAb,QAAA"}