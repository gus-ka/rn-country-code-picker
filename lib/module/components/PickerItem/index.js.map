{"version":3,"names":["React","TouchableOpacity","View","Flag","StyledText","styles","PickerItem","_ref","country","onCountrySelect","textStyle","containerStyle","name","flag","callingCode","createElement","onPress","testID","style","container","imageContainer","codeContainer"],"sources":["index.tsx"],"sourcesContent":["import React from 'react';\nimport { TouchableOpacity, View } from 'react-native';\nimport { IPickerItemProps } from '../../types';\nimport Flag from '../Flag';\nimport StyledText from '../StyledText';\nimport styles from './styles';\n\nconst PickerItem: React.FC<IPickerItemProps> = ({\n  country,\n  onCountrySelect,\n  textStyle,\n  containerStyle,\n}) => {\n  const { name, flag, callingCode } = country;\n  return (\n    <TouchableOpacity onPress={() => onCountrySelect(country)} testID=\"picker-item\">\n      <View style={containerStyle ?? styles.container}>\n        <View style={styles.imageContainer}>\n          <Flag flag={flag} />\n        </View>\n        <View style={styles.codeContainer}>\n          <StyledText style={textStyle}>{`+${callingCode}`}</StyledText>\n        </View>\n        <StyledText style={textStyle ?? styles.textStyle}>{name}</StyledText>\n      </View>\n    </TouchableOpacity>\n  );\n};\n\nexport default PickerItem;\n"],"mappings":"AAAA,OAAOA,KAAK,MAAM,OAAO;AACzB,SAASC,gBAAgB,EAAEC,IAAI,QAAQ,cAAc;AAErD,OAAOC,IAAI,MAAM,SAAS;AAC1B,OAAOC,UAAU,MAAM,eAAe;AACtC,OAAOC,MAAM,MAAM,UAAU;AAE7B,MAAMC,UAAsC,GAAGC,IAAA,IAKzC;EAAA,IAL0C;IAC9CC,OAAO;IACPC,eAAe;IACfC,SAAS;IACTC;EACF,CAAC,GAAAJ,IAAA;EACC,MAAM;IAAEK,IAAI;IAAEC,IAAI;IAAEC;EAAY,CAAC,GAAGN,OAAO;EAC3C,oBACER,KAAA,CAAAe,aAAA,CAACd,gBAAgB;IAACe,OAAO,EAAEA,CAAA,KAAMP,eAAe,CAACD,OAAO,CAAE;IAACS,MAAM,EAAC;EAAa,gBAC7EjB,KAAA,CAAAe,aAAA,CAACb,IAAI;IAACgB,KAAK,EAAEP,cAAc,IAAIN,MAAM,CAACc;EAAU,gBAC9CnB,KAAA,CAAAe,aAAA,CAACb,IAAI;IAACgB,KAAK,EAAEb,MAAM,CAACe;EAAe,gBACjCpB,KAAA,CAAAe,aAAA,CAACZ,IAAI;IAACU,IAAI,EAAEA;EAAK,CAAE,CACf,CAAC,eACPb,KAAA,CAAAe,aAAA,CAACb,IAAI;IAACgB,KAAK,EAAEb,MAAM,CAACgB;EAAc,gBAChCrB,KAAA,CAAAe,aAAA,CAACX,UAAU;IAACc,KAAK,EAAER;EAAU,GAAG,IAAGI,WAAY,EAAc,CACzD,CAAC,eACPd,KAAA,CAAAe,aAAA,CAACX,UAAU;IAACc,KAAK,EAAER,SAAS,IAAIL,MAAM,CAACK;EAAU,GAAEE,IAAiB,CAChE,CACU,CAAC;AAEvB,CAAC;AAED,eAAeN,UAAU"}