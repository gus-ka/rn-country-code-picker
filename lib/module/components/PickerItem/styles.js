import { StyleSheet } from 'react-native';
import { spacing } from '../../theme';
export default StyleSheet.create({
  container: {
    flexDirection: 'row',
    padding: spacing.s
  },
  imageContainer: {
    alignSelf: 'center'
  },
  codeContainer: {
    alignSelf: 'center'
  },
  textStyle: {
    alignSelf: 'center'
  }
});
//# sourceMappingURL=styles.js.map