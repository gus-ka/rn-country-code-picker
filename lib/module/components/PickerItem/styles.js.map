{"version":3,"names":["StyleSheet","spacing","create","container","flexDirection","padding","s","imageContainer","alignSelf","codeContainer","textStyle"],"sources":["styles.ts"],"sourcesContent":["import { StyleSheet } from 'react-native';\nimport { colors, spacing } from '../../theme';\n\nexport default StyleSheet.create({\n  container: {\n    flexDirection: 'row',\n    padding: spacing.s,\n  },\n  imageContainer: {\n    alignSelf: 'center',\n  },\n  codeContainer: {\n    alignSelf: 'center',\n  },\n  textStyle: { alignSelf: 'center' },\n});\n"],"mappings":"AAAA,SAASA,UAAU,QAAQ,cAAc;AACzC,SAAiBC,OAAO,QAAQ,aAAa;AAE7C,eAAeD,UAAU,CAACE,MAAM,CAAC;EAC/BC,SAAS,EAAE;IACTC,aAAa,EAAE,KAAK;IACpBC,OAAO,EAAEJ,OAAO,CAACK;EACnB,CAAC;EACDC,cAAc,EAAE;IACdC,SAAS,EAAE;EACb,CAAC;EACDC,aAAa,EAAE;IACbD,SAAS,EAAE;EACb,CAAC;EACDE,SAAS,EAAE;IAAEF,SAAS,EAAE;EAAS;AACnC,CAAC,CAAC"}