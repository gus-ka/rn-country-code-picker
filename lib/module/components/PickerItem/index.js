import React from 'react';
import { TouchableOpacity, View } from 'react-native';
import Flag from '../Flag';
import StyledText from '../StyledText';
import styles from './styles';
const PickerItem = _ref => {
  let {
    country,
    onCountrySelect,
    textStyle,
    containerStyle
  } = _ref;
  const {
    name,
    flag,
    callingCode
  } = country;
  return /*#__PURE__*/React.createElement(TouchableOpacity, {
    onPress: () => onCountrySelect(country),
    testID: "picker-item"
  }, /*#__PURE__*/React.createElement(View, {
    style: containerStyle ?? styles.container
  }, /*#__PURE__*/React.createElement(View, {
    style: styles.imageContainer
  }, /*#__PURE__*/React.createElement(Flag, {
    flag: flag
  })), /*#__PURE__*/React.createElement(View, {
    style: styles.codeContainer
  }, /*#__PURE__*/React.createElement(StyledText, {
    style: textStyle
  }, `+${callingCode}`)), /*#__PURE__*/React.createElement(StyledText, {
    style: textStyle ?? styles.textStyle
  }, name)));
};
export default PickerItem;
//# sourceMappingURL=index.js.map