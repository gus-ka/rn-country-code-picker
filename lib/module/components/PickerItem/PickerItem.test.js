import React from 'react';
import { fireEvent, render } from '@testing-library/react-native';
import mockCountry from '../../__mocks__/country';
import PickerItem from './';
const onCountrySelectMockFn = jest.fn();
const PICKER_ITEM_COMPONENT = /*#__PURE__*/React.createElement(PickerItem, {
  country: mockCountry,
  onCountrySelect: onCountrySelectMockFn
});
test('renders correctly', () => {
  const {
    toJSON,
    getByText
  } = render(PICKER_ITEM_COMPONENT);
  const rendered = toJSON();
  const name = getByText(`${mockCountry.name} `);
  const callingCode = getByText(`+${mockCountry.callingCode} `);
  expect(name).toBeTruthy();
  expect(callingCode).toBeTruthy();
  expect(rendered).toMatchSnapshot();
});
test('picks country on click', () => {
  const {
    getByTestId
  } = render(PICKER_ITEM_COMPONENT);
  const pickerItem = getByTestId('picker-item');
  fireEvent.press(pickerItem);
  expect(onCountrySelectMockFn).toHaveBeenCalledWith(mockCountry);
});
//# sourceMappingURL=PickerItem.test.js.map