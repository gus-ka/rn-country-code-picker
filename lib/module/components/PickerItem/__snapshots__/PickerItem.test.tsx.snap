// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID="picker-item"
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "padding": 8,
      }
    }
  >
    <View
      style={
        Object {
          "alignSelf": "center",
        }
      }
    >
      <View
        style={
          Object {
            "borderRadius": 4,
            "height": 30,
            "marginRight": 8,
            "width": 40,
          }
        }
      >
        <Image
          resizeMethod="scale"
          source={
            Object {
              "testUri": "../../../src/assets/flags/tr.png",
            }
          }
          style={
            Object {
              "height": "100%",
              "width": "100%",
            }
          }
          testID="flag-image"
        />
      </View>
    </View>
    <View
      style={
        Object {
          "alignSelf": "center",
        }
      }
    >
      <Text
        style={
          Array [
            undefined,
            Object {
              "fontSize": 16,
              "lineHeight": 24,
            },
          ]
        }
      >
        +90
         
      </Text>
    </View>
    <Text
      style={
        Array [
          Object {
            "alignSelf": "center",
          },
          Object {
            "fontSize": 16,
            "lineHeight": 24,
          },
        ]
      }
    >
      Turkey
       
    </Text>
  </View>
</View>
`;
